services/dynamic-scheduler/tests/unit/status_monitor/test_services_status_monitor__monitor.py

```diff
@@ -0,0 +1,409 @@
+) -> None:
+    async for attempt in AsyncRetrying(
+        reraise=True,
+        stop=stop_after_delay(1),
+        wait=wait_fixed(0.01),
+        retry=retry_if_exception_type(AssertionError),
+    ):
+        with attempt:
+
+            assert deferred_status_spies["on_result"].call_count == len(timeline)
+            assert [
+                x.args[0] for x in deferred_status_spies["on_result"].call_args_list
+            ] == timeline
+
+
+async def _assert_notification_count(
+    mock: AsyncMock, expected_count: NonNegativeInt
+) -> None:
+    async for attempt in AsyncRetrying(
+        reraise=True,
+        stop=stop_after_delay(1),
+        wait=wait_fixed(0.01),
+        retry=retry_if_exception_type(AssertionError),
+    ):
+        with attempt:
+            assert mock.call_count == expected_count
+
+
+@pytest.fixture
+async def mock_director_v2_status(
+    app: FastAPI, response_timeline: _ResponseTimeline
+) -> AsyncIterable[None]:
+    def _side_effect_node_status_response(request: Request) -> Response:
+        node_id = NodeID(f"{request.url}".split("/")[-1])
+
+        service_status = response_timeline.get_status(node_id)
+
+        if isinstance(service_status, NodeGet):
+            return Response(
+                status.HTTP_200_OK,
+                text=json.dumps(jsonable_encoder({"data": service_status.dict()})),
+            )
+        if isinstance(service_status, DynamicServiceGet):
+            return Response(status.HTTP_200_OK, text=service_status.json())
+        if isinstance(service_status, NodeGetIdle):
+            return Response(status.HTTP_404_NOT_FOUND)
+
+        raise TypeError
+
+    with respx.mock(
+        base_url=app.state.settings.DYNAMIC_SCHEDULER_DIRECTOR_V2_SETTINGS.api_base_url,
+        assert_all_called=False,
+        assert_all_mocked=True,
+    ) as mock:
+        mock.get(re.compile(r"/dynamic_services/([\w-]+)")).mock(
+            side_effect=_side_effect_node_status_response
+        )
+        yield
+
+
+@pytest.fixture
+def monitor(mock_director_v2_status: None, app: FastAPI) -> Monitor:
+    return get_monitor(app)
+
+
+@pytest.fixture
+def deferred_status_spies(mocker: MockerFixture) -> dict[str, AsyncMock]:
+    results: dict[str, AsyncMock] = {}
+    for method_name in (
+        "start",
+        "on_result",
+        "on_created",
+        "run",
+        "on_finished_with_error",
+    ):
+        mock_method = mocker.AsyncMock(wraps=getattr(DeferredGetStatus, method_name))
+        mocker.patch.object(DeferredGetStatus, method_name, mock_method)
+        results[method_name] = mock_method
+
+    return results
+
+
+@pytest.fixture
+def remove_tracked_spy(mocker: MockerFixture) -> AsyncMock:
+    mock_method = mocker.AsyncMock(wraps=_monitor.service_tracker.remove_tracked)
+    return mocker.patch.object(_monitor.service_tracker, "remove_tracked", mock_method)
+
+
+@pytest.fixture
+def node_id() -> NodeID:
+    return _DEFAULT_NODE_ID
```

Comment:
not sure I get the need for this

Reply:
Please search the uses of `_DEFAULT_NODE_ID`. It is used outside of the fixture context to generate data for parametrising the below test